src/main/java/io/codelex/advance_exercise/first_exercise/CreditCard.java

```diff
@@ -0,0 +1,40 @@
+package io.codelex.advance_exercise.first_exercise;
+
+import java.math.BigDecimal;
+
+public class CreditCard extends Card {
+    public CreditCard(int number, String fullName, int CCV, BigDecimal balance) {
+        super(number, fullName, CCV, balance);
+    }
+
+    @Override
+    public void withdraw(BigDecimal amount) {
+        BigDecimal balance = super.getBalance();
+        super.setBalance(balance.add(amount));
+    }
+
+
+    @Override
+    public void deposit(BigDecimal amount) throws NotEnoughFundsException {
+        BigDecimal balance = super.getBalance();
+        if (balance.compareTo(amount) == -1) {
+            throw new NotEnoughFundsException("It's not possible!");
```

Comment:
Šis gan joprojām neizskatās ok, ka metode deposit met not enough funds exception. Jo deposit jau nozīmē naudu pievienot.